ml4gw/dataloading/hdf5_dataset.py

```diff
@@ -0,0 +1,175 @@
+import warnings
+from typing import Sequence, Union
+
+import h5py
+import numpy as np
+import torch
+
+from ml4gw.types import WaveformTensor
+
+
+class ContiguousHdf5Warning(Warning):
+    pass
+
+
+class Hdf5TimeSeriesDataset(torch.utils.data.IterableDataset):
+    """
+    Iterable dataset that samples and loads windows of
+    timeseries data uniformly from a set of HDF5 files.
+    It is _strongly_ recommended that these files have been
+    written using [chunked storage]
+    (https://docs.h5py.org/en/stable/high/dataset.html#chunked-storage).
+    This has shown to produce increases in read-time speeds
+    of over an order of magnitude.
+
+    Args:
+        fnames:
+            Paths to HDF5 files from which to sample data.
+        channels:
+            Datasets to read from the indicated files, which
+            will be stacked along dim 1 of the generated batches
+            during iteration.
+        kernel_size:
+            Size of the windows to read, in number of samples.
+            This will be the size of the last dimension of the
+            generated batches.
+        batch_size:
+            Number of windows to sample at each iteration.
+        batches_per_epoch:
+            Number of batches to generate during each call
+            to `__iter__`.
+        coincident:
+            Whether windows for each channel in a given batch
+            element should be sampled coincidentally, i.e.
+            corresponding to the same time indices from the
+            same files, or should be sampled independently.
+            For the latter case, users can either specify
+            `False`, which will sample filenames independently
+            for each channel, or `"files"`, which will sample
+            windows independently within a given file for each
+            channel. The latter setting limits the amount of
+            entropy in the effective dataset, but can provide
+            over 2x improvement in total throughput.
+    """
+
+    def __init__(
+        self,
+        fnames: Sequence[str],
+        channels: Sequence[str],
+        kernel_size: int,
+        batch_size: int,
+        batches_per_epoch: int,
+        coincident: Union[bool, str],
+    ) -> None:
+        if not isinstance(coincident, bool) and coincident != "files":
+            raise ValueError(
+                "coincident must be either a boolean or 'file', "
```

Comment:
error message should say "files" not "file"

Reply:
Yup good catch